psci/PSCi.hs

```diff
@@ -486,9 +486,24 @@ handleBrowse moduleName = do
   case env of
     Left errs -> printErrors errs
     Right env' ->
-      if moduleName `notElem` (nub . map ((\ (P.Module _ _ modName _ _ ) -> modName) . snd)) (psciLoadedModules st)
-        then PSCI $ outputStrLn $ "Module '" ++ N.runModuleName moduleName ++ "' is not valid."
-        else printModuleSignatures moduleName env'
+      if isModInEnv moduleName st
+        then printModSigs moduleName env'
+        else case lookupUnQualifiedModName moduleName st of
+          Just unQualifiedName ->
+            if isModInEnv unQualifiedName st
+              then printModSigs unQualifiedName env'
+              else failNotInEnv moduleName
+          Nothing ->
+            failNotInEnv moduleName
+  where
+    isModInEnv modName st =
+        modName `elem` (nub . map ((\ (P.Module _ _ mName _ _ ) -> mName) . snd)) (psciLoadedModules st)
```

Comment:
Perhaps use `find` instead of `lookup` with `map`. The `nub` also seems redundant.